packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1383,6 +1414,10 @@ export class Domain extends DomainBase implements IDomain {
       throw new Error('The t2.micro.elasticsearch instance type supports only Elasticsearch 1.5 and 2.3.');
     }
 
+    if (isSomeInstanceType('t2', 't3') && warmEnabled) {
+      throw new Error('T2 and T3 instance types do not support UltraWarm storage.');
```

Comment:
Can you add a documentation link for this limitation? or an example of the CFN failure?

Reply:
It is described in the documentation: https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/aes-supported-instance-types.html
> The T2 instance types do not support ..., UltraWarm storage, ...
> The T3 instance types do not support UltraWarm storage.

The link to the documentation is also mentioned as a comment in the code.

Reply:
I might be misunderstanding something, but to me this documentation means that you cannot configure `T2` and `T3` types for the type of the `UltraWarm` nodes, but you can still use them for other nodes. Is that not the case?

Reply:
UltraWarm uses special instance types that are beginning with 'ultrawarm'. If you use another instance type like `T3`, a CloudFormation error appears:

> 1 validation error detected: Value 't3.small.elasticsearch' at 'elasticsearchClusterConfig.warmType' failed to satisfy constraint: Member must satisfy enum value set: [ultrawarm1.xlarge.elasticsearch, ultrawarm1.medium.elasticsearch, ultrawarm1.large.elasticsearch]

I just added this restriction to the validations.

I used a valid type for the UltraWarm node and `T3` for the instance type. The CloudFormation error message says, that it is not supported in this case:

> Warm storage is not available for the t3.small.elasticsearch instance type. Select a different instance type.

Reply:
👍